ServerCore/Pages/Rooms/RoomList.cshtml.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Rooms
+{
+    [Authorize(Policy = "IsEventAdmin")]
+    public class RoomListModel : EventSpecificPageModel
+    {
+        public RoomListModel(PuzzleServerContext serverContext, UserManager<IdentityUser> manager) : base(serverContext, manager)
+        {
+        }
+
+        public IList<Room> Room { get; set; }
+        public IList<Tuple<string, bool>> Groups { get; set; }
+        private Dictionary<string, bool> groupMapping = new Dictionary<string, bool>();
+        public bool AssignRooms { get; set; }
+
+        public async Task OnGetAsync()
+        {
+            await FillCollections();
+        }
+
+        public async Task<IActionResult> OnPostAsync(List<string> GroupToToggle)
+        {
+            await FillCollections();
+
+            // Update the rooms that are online if that's changed
+
+            foreach (Tuple<string, bool> gr in Groups)
+            {
+                if (GroupToToggle.Contains(gr.Item1) && !gr.Item2)
+                {
+                    groupMapping.Add(gr.Item1, true);
+                }
+                else if(!GroupToToggle.Contains(gr.Item1) && gr.Item2)
+                {
+                    // TODO: Add notification or something for rooms going offline that have been assigned to teams
```

Comment:
Do we need this for this week?

Reply:
Probably not, but if we don't have this then the rooms & groups can get out of sync (which should also be ok since I'm just overwriting when there's a difference, but it's messy)

Reply:
Oh, do you mean the notification? I can put that in another PR, but right now there isn't any way to take a specific conference room offline with the UI so we're going to want to handle things manually anyway during the event.